rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -56,19 +59,112 @@
 #include "rosidl_typesupport_introspection_cpp/service_introspection.hpp"
 #include "rosidl_typesupport_introspection_cpp/visibility_control.h"
 
+#include "rosidl_typesupport_introspection_c/field_types.h"
+#include "rosidl_typesupport_introspection_c/identifier.h"
+#include "rosidl_typesupport_introspection_c/message_introspection.h"
+#include "rosidl_typesupport_introspection_c/service_introspection.h"
+#include "rosidl_typesupport_introspection_c/visibility_control.h"
+
 #include "rmw_connext_shared_cpp/shared_functions.hpp"
 #include "rmw_connext_shared_cpp/types.hpp"
 
+#include "macros.hpp"
+
+
+bool using_introspection_c_typesupport(const char * typesupport_identifier)
+{
+  return strcmp(typesupport_identifier, rosidl_typesupport_introspection_c__identifier) == 0;
```

Comment:
The should not use string comparison since it is a pretty expensive operation. See the previously used macro: https://github.com/ros2/rmw/blob/55254b66ddb0f7c5ff8f05fb8c0c60378fb3b2da/rmw/include/rmw/impl/cpp/macros.hpp#L70

Same below.